engine/access/rpc/backend/backend_accounts.go

```diff
@@ -80,6 +80,114 @@ func (b *backendAccounts) GetAccountAtBlockHeight(
 	return account, nil
 }
 
+// GetAccountBalanceAtLatestBlock returns the account balance at the latest sealed block.
+func (b *backendAccounts) GetAccountBalanceAtLatestBlock(ctx context.Context, address flow.Address) (uint64, uint64, error) {
+	sealed, err := b.state.Sealed().Head()
+	if err != nil {
+		err := irrecoverable.NewExceptionf("failed to lookup sealed header: %w", err)
+		irrecoverable.Throw(ctx, err)
+		return 0, 0, err
+	}
+
+	sealedBlockID := sealed.ID()
+
+	account, availableBalance, err := b.getAccountBalanceAtBlock(ctx, address, sealedBlockID, sealed.Height)
+	if err != nil {
+		b.log.Debug().Err(err).Msgf("failed to get account balance at blockID: %v", sealedBlockID)
+		return 0, 0, err
+	}
+
+	return account, availableBalance, nil
+}
+
+// GetAccountBalanceAtBlockHeight returns the account balance at the given block height.
+func (b *backendAccounts) GetAccountBalanceAtBlockHeight(
+	ctx context.Context,
+	address flow.Address,
+	height uint64,
+) (uint64, uint64, error) {
+	blockID, err := b.headers.BlockIDByHeight(height)
+	if err != nil {
+		return 0, 0, rpc.ConvertStorageError(err)
+	}
+
+	account, availableBalance, err := b.getAccountBalanceAtBlock(ctx, address, blockID, height)
+	if err != nil {
+		b.log.Debug().Err(err).Msgf("failed to get account balance at height: %v", height)
+		return 0, 0, err
+	}
+
+	return account, availableBalance, nil
+}
+
+// GetAccountKeysAtLatestBlock returns the account public keys at the latest sealed block.
+func (b *backendAccounts) GetAccountKeysAtLatestBlock(
+	ctx context.Context,
+	address flow.Address,
+	keyIndex int,
+) ([]flow.AccountPublicKey, error) {
+	sealed, err := b.state.Sealed().Head()
+	if err != nil {
+		err := irrecoverable.NewExceptionf("failed to lookup sealed header: %w", err)
+		irrecoverable.Throw(ctx, err)
+		return nil, err
+	}
+
+	sealedBlockID := sealed.ID()
+
+	var accountKeys []flow.AccountPublicKey
+	if keyIndex == -1 {
```

Comment:
Not sure I like this special casing. I would prefer two separate methods:
- GetAccountKeysAtLatestBlock
- GetAccountKeyAtLatestBlock